### PR TITLE
Introduction to Binary Trees

### LINKED ISSUE #78: [Feature Request] Binary Trees
### Is your feature request related to a problem?
Binary trees are a fundamental data structure with numerous applications in computer science, including expression parsing, sorting algorithms, and hierarchical data representation. Their absence from our current offerings limits our ability to solve problems that benefit from efficient data organization and retrieval.

## Describe the solution you'd like 
An intro to Binary trees should be provided including their properties, types, advantages, and disadvantages.

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by KhawajaFashi**
### Description
This pull request introduces an Introduction to Binary Trees section, covering the basic concepts, properties, types, and advantages/disadvantages of binary trees. This content is aimed to enhance the documentation and provide readers with foundational knowledge about binary trees. The section explains:

- Definition and structure of binary trees (nodes, edges).
- Key properties (height, depth, balance).
- Different types of binary trees (Full, Complete, Perfect, Balanced, Degenerate).
- Advantages and disadvantages.

Fixes #78 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Binary Trees/binary-tree.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+- **Right Child:** A reference to the right subtree (or null if no right child exists).
+
+The tree begins with a single node called the **root**. The hierarchical structure allows for organized storage and retrieval of data.
+
+## Properties
+Key characteristics of binary trees include:
+- **Height:** The length of the longest path from the root to a leaf node. The height of an empty tree is -1, and the height of a tree with only one node is 0.
+- **Depth:** The distance from the root to a specific node. The root node has a depth of 0.
+- **Balance:** A tree is considered balanced if the heights of the left and right subtrees of any node differ by at most one.
+    <pre>         A
```
Comment (ajay-dhangar) [inline docs/Binary Trees/binary-tree.md]: replace

```md
    <pre>
           A
         /   \
        B     C
       /     / \
      D     F   G
    </pre>
```

with

````md
```
          A
         /   \
        B     C
       /     / \
      D     F   G
```
````

File: docs/Binary Trees/binary-tree.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    **Height of the tree**: 2
+
+    **Depth of D, E, F**: 2
+
+    **Balanced**: Yes, the tree is balanced.
+
+
+## Types of Binary Trees
+1. **Full Binary Trees:** Every node has either 0 or 2 children.
+    <pre>      A
```
Comment (ajay-dhangar) [inline docs/Binary Trees/binary-tree.md]: replace `<pre>` with ```` ``` ```` (following  .md / .mdx syntex )

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
@KhawajaFashi Your code needs to be updated based on the suggestions provided in the comments. Please let me know if you have any questions.

File: docs/Binary Trees/binary-tree.md (hunk 2 above)
Comment (KhawajaFashi) [reply docs/Binary Trees/binary-tree.md]: Ok sir I will change it accordingly

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**